### PR TITLE
Core: Final API renaming for 5.12

### PR DESCRIPTION
**Pull request by KerstinKeller**
GetTopicInformation -> GetDataTypeInformation
SetTopicInformation -> SetDataTypeInformation
SDataTypeInformation

**Pull request type**
API changes for 5.12 (only for types introduced since 5.11).

### CODE REVIEW

File: ecal/core/src/service/ecal_clientgate.cpp [hunk 1]
```diff
@@ -171,21 +178,19 @@ namespace eCAL
 
   bool CClientGate::ApplyServiceToDescGate(const std::string& service_name_
     , const std::string& method_name_
-    , const std::string& req_type_name_
-    , const std::string& req_type_desc_
-    , const std::string& resp_type_name_
-    , const std::string& resp_type_desc_)
+    , const SDataTypeInformation& reqest_type_description_
+    , const SDataTypeInformation& response_type_description_)
```
Comment (FlorianReimold) [inline ecal/core/src/service/ecal_clientgate.cpp:182]:
```suggestion
    , const SDataTypeInformation& reqest_type_information_
    , const SDataTypeInformation& response_type_information_)
```

File: ecal/core/src/service/ecal_clientgate.cpp [hunk 2]
```diff
@@ -171,21 +178,19 @@ namespace eCAL
...
   {
     if (g_descgate() != nullptr)
     {
       // calculate the quality of the current info
       ::eCAL::CDescGate::QualityFlags quality = ::eCAL::CDescGate::QualityFlags::NO_QUALITY;
-      if (!(req_type_name_.empty() && resp_type_name_.empty()))
+      if (!(reqest_type_description_.name.empty() && response_type_description_.name.empty()))
```
Comment (FlorianReimold) [inline ecal/core/src/service/ecal_clientgate.cpp:188]:
```suggestion
      if (!(reqest_type_information_.name.empty() && response_type_information_.name.empty()))
```

File: ecal/core/src/service/ecal_clientgate.cpp [hunk 3]
```diff
@@ -171,21 +178,19 @@ namespace eCAL
...
         quality |= ::eCAL::CDescGate::QualityFlags::TYPE_AVAILABLE;
-      if (!(req_type_desc_.empty() && resp_type_desc_.empty()))
+      if (!(reqest_type_description_.descriptor.empty() && response_type_description_.descriptor.empty()))
```
Comment (FlorianReimold) [inline ecal/core/src/service/ecal_clientgate.cpp:190]:
```suggestion
      if (!(reqest_type_information_.descriptor.empty() && response_type_information_.descriptor.empty()))
```

File: ecal/core/src/service/ecal_clientgate.cpp [hunk 4]
```diff
@@ -171,21 +178,19 @@ namespace eCAL
...
         quality |= ::eCAL::CDescGate::QualityFlags::DESCRIPTION_AVAILABLE;
 
-      return g_descgate()->ApplyServiceDescription(service_name_, method_name_, req_type_name_, req_type_desc_, resp_type_name_, resp_type_desc_, quality);
+      return g_descgate()->ApplyServiceDescription(service_name_, method_name_, reqest_type_description_, response_type_description_, quality);
```
Comment (FlorianReimold) [inline ecal/core/src/service/ecal_clientgate.cpp:193]:
```suggestion
      return g_descgate()->ApplyServiceDescription(service_name_, method_name_, reqest_type_information_, response_type_information_, quality);
```

File: app/mon/mon_cli/src/ecal_mon_cli.cpp [hunk 5]
```diff
@@ -352,9 +352,9 @@ void ProcProto(const std::string& topic_name, int msg_count)
...
 
   // get topic type
-  eCAL::STopicInformation topic_info;
-  eCAL::Util::GetTopicInformation(topic_name, topic_info);
-  if(topic_info.type.empty())
+  eCAL::SDataTypeInformation topic_info;
+  eCAL::Util::GetTopicDataTypeInformation(topic_name, topic_info);
+  if(topic_info.name.empty())
   {
     std::cout << "could not get type for topic " << topic_name << std::endl;
```
[the remark this answers is not in the document]
Comment (KerstinKeller) [reply app/mon/mon_cli/src/ecal_mon_cli.cpp:359]: I don't agree 100%. Maybe "could not get typename for topic" -> it's not the topic name we're return here, but the name of the topic type.

File: ecal/core/include/ecal/ecal_util.h [hunk 6]
```diff
@@ -278,32 +271,32 @@ namespace eCAL
...
     **/
-    ECAL_DEPRECATE_SINCE_5_13("Please use the method bool GetTopicInformation(const std::string& topic_name_, STopicInformation& topic_info_) instead. You can extract the type information from the members encoding and type of the STopicInformation variable. This function will be removed in eCAL6.")
+    ECAL_DEPRECATE_SINCE_5_13("Please use the method bool GetDataTypeInformation(const std::string& topic_name_, SDataTypeInformation& topic_info_) instead. You can extract the type information from the members encoding and type of the STopicInformation variable. This function will be removed in eCAL6.")
     ECAL_API std::string GetTypeName(const std::string& topic_name_);
 
     /**
      * @brief Gets description of the specified topic.
      *
-     * This function is deprecated with eCAL 5.10. Please use GetTopicInformation.
+     * This function is deprecated with eCAL 5.10. Please use GetTopicDataTypeInformation.
```
[the remark this answers is not in the document]
Comment (KerstinKeller) [reply ecal/core/include/ecal/ecal_util.h:280]: you are right, they have been deprecated a longer time ago, but now the functions to use instead have also been deprecated. i corrected this.

[the remark this answers is not in the document]
Comment (FlorianReimold) [reply ecal/core/include/ecal/ecal_util.h:280]: Yes, but please scroll down a little bit. There it says ECAL_DEPRECATE_SINCE_5_13, while the comment says 5.10

File: ecal/core/include/ecal/ecal_util.h [hunk 7]
```diff
@@ -278,32 +271,32 @@ namespace eCAL
...
     **/
-    ECAL_DEPRECATE_SINCE_5_13("Please use the method bool GetTopicInformation(const std::string& topic_name_, STopicInformation& topic_info_) instead. You can extract the type information from the members encoding and type of the STopicInformation variable. This function will be removed in eCAL6.")
+    ECAL_DEPRECATE_SINCE_5_13("Please use the method bool GetDataTypeInformation(const std::string& topic_name_, SDataTypeInformation& topic_info_) instead. You can extract the type information from the members encoding and type of the STopicInformation variable. This function will be removed in eCAL6.")
     ECAL_API bool GetDescription(const std::string& topic_name_, std::string& topic_desc_);
 
     /**
      * @brief Gets description of the specified topic.
      *
-     * This function is deprecated with eCAL 5.10. Please use GetTopicInformation.
+     * This function is deprecated with eCAL 5.10. Please use GetTopicDataTypeInformation.
```
[the remark this answers is not in the document]
Comment (FlorianReimold) [reply ecal/core/include/ecal/ecal_util.h:293]: Yes, but please scroll down a little bit. There it says ECAL_DEPRECATE_SINCE_5_13, while the comment says 5.10

**Pull request closed without merging**